### PR TITLE
Add component explanation to kubeconfig-file.md

### PR DESCRIPTION
**Pull request by j3ffml**
cc @jayunit100

### CODE REVIEW

**Comment by jayunit100** [comment]
Looks like some reasonable updates to me.  

This still seems very abstract .  

I guess a diagram might be useful in the future. But good to get this info merged in for now.

**Comment by jlowdermilk** [comment]
ping @RichieEscarez

**Comment by RichieEscarez** [comment]
@jlowdermilk sorry, i let this get lost in my inbox. I might not be able to get to this. Ill let @jayunit100 take over.

**Comment by jlowdermilk** [comment]
@kubernetes/kubectl can someone glance over this and add the tag if it looks good?

File: docs/user-guide/kubeconfig-file.md [hunk 1]
```diff
@@ -94,7 +94,89 @@ users:
...
+clusters:
+- cluster:
+    api-version: v1
+    server: http://cow.org:8080
+  name: cow-cluster
+```
+
+A `cluster` contains endpoint data for a kubernetes cluster. This includes the fully
+qualified url for the kubernetes apiserver, as well as the cluster's certificate
+authority or `insecure-skip-tls-verify: true`, if the cluster does not support tls.
```
Comment (janetkuo) [inline docs/user-guide/kubeconfig-file.md]: Suggest "... certificate authority specified in `certificate-authority`, or `insecure-skip-tls-verify: true` if the cluster does not support TLS."

Perhaps include `certificate-authority` and/or `insecure-skip-tls-verify: true` in the example here would make it more clear?

File: docs/user-guide/kubeconfig-file.md [hunk 2]
```diff
@@ -44,15 +44,15 @@ So in order to easily switch between multiple clusters, for multiple users, a ku
...
 
 ## Related discussion
 
 http://issue.k8s.io/1755
 
-## Example kubeconfig file
+## Components of a kubeconfig file
 
-The below file contains a `current-context` which will be used by default by clients which are using the file to connect to a cluster.  Thus, this kubeconfig file has more information in it then we will necessarily have to use in a given session.  You can see it defines many clusters, and users associated with those clusters.  The context itself is associated with both a cluster AND a user.
+### Example kubeconfig file
```
Comment (janetkuo) [inline docs/user-guide/kubeconfig-file.md]: We can mention somewhere in this document that `kubectl config view` shows you the merged kubeconfig settings.

File: docs/user-guide/kubeconfig-file.md [hunk 3]
```diff
@@ -94,7 +94,89 @@ users:
...
+- name: green-user
+  user:
+    client-certificate: path/to/my/client/cert
+    client-key: path/to/my/client/key
+```
+
+A `user` defines client credentials for authenticating to a kubernetes cluster. A
+`user` has a name (nickname) which acts as its key within the list of user entries
+after kubeconfig is loaded/merged. You can add or modify `user` entries using
+[`kubectl config set-credentials`](kubectl/kubectl_config_set-credentials.md).
```
Comment (janetkuo) [inline docs/user-guide/kubeconfig-file.md]: Mention available credentials here? (`client-certificate`, `client-key`, `token`, `username` & `password`)

**Comment by janetkuo** [comment]
This looks good. I have a few more suggestions.

File: docs/user-guide/kubeconfig-file.md (hunk 1 above)
Comment (deads2k) [inline docs/user-guide/kubeconfig-file.md]: "if the cluster does not support tls." isn't quite right.  "If the cluster's serving certificate is not signed by a system trusted certificate authority."?

File: docs/user-guide/kubeconfig-file.md (hunk 3 above)
Comment (deads2k) [inline docs/user-guide/kubeconfig-file.md]: Also,  username/password and token are mutually exclusive, but client-certs and keys can be combined with them.

**Comment by deads2k** [comment]
Minor comments

File: docs/user-guide/kubeconfig-file.md (hunk 2 above)
Comment (jlowdermilk) [inline docs/user-guide/kubeconfig-file.md]: Added a short section on viewing kubeconfig.

**Comment by jlowdermilk** [comment]
ptal

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

**Comment by deads2k** [comment]
lgtm

**Comment by janetkuo** [comment]
LGTM

**Comment by jlowdermilk** [comment]
@k8s-bot unit test this

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**